### PR TITLE
Deploy Container Registry Mirror again

### PR DESCRIPTION
**Pull request by zoetrope**
Signed-off-by: zoetrope <a.ikezoe@gmail.com>

### CODE REVIEW

**Comment by zoetrope** [comment]
wait for https://github.com/cybozu-go/neco/pull/1923

File: monitoring/base/victoriametrics/rules/registry-scrape.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    matchLabels:
+      app.kubernetes.io/name: registry
+  podMetricsEndpoints:
+    - port: metrics
+      relabelConfigs:
+        - replacement: registry
+          targetLabel: job
+        - sourceLabels: [ __meta_kubernetes_pod_label_app_kubernetes_io_name ]
+          action: replace
+          targetLabel: registry
```
Comment (umezawatakeshi) [inline monitoring/base/victoriametrics/rules/registry-scrape.yaml:19]: Is is correct?
I think that this relabeling is intended to set upstream registry name (such as `quay` or `ghcr`) to `registry` metrics label but `__meta_kubernetes_pod_label_app_kubernetes_io_name` (that is, `app.kubernetes.io/name` pod label) has fixed value `registry`.

File: registry/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,102 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: registry-{{ .Values.instance }}
+  namespace: registry
+  labels:
+    app.kubernetes.io/name: registry
+    app.kubernetes.io/instance: {{ .Values.instance }}
```
Comment (umezawatakeshi) [inline registry/templates/deployment.yaml:8]: Isn't `app.kubernetes.io/instance` label overwritten by Argo CD ?

File: monitoring/base/victoriametrics/rules/registry-alertrule.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+metadata:
+  name: registry
+  namespace: monitoring
+spec:
+  groups:
+    - name: registry
+      rules:
+        - alert: RegistryDown
+          expr: |
+            absent(up{job="registry"} == 1)
```
Comment (umezawatakeshi) [inline monitoring/base/victoriametrics/rules/registry-alertrule.yaml:12]: This rule alerts only if all registry mirror pods for all upstream registry are down. If all pods for certain upstream are down and some pods for other upstream are up, this alert won't be fired.

Is it intended?

**Review by umezawatakeshi — APPROVED** [review]

**Comment by zoetrope** [comment]
@umezawatakeshi 
Please approve it again. Conflict fixed.

**Review by umezawatakeshi — APPROVED** [review]

**Pull request merged**